extensions/query-engine/src/main/java/com/aerospike/helper/query/QueryEngine.java

```diff
@@ -650,31 +641,8 @@ public Namespace getNamespace(String namespace) {
 	/**
 	 * refreshes the Index cache from the Cluster
 	 */
-	public synchronized void refreshIndexes() {
-		/*
-		 * cache index by Bin name
-		 */
-		if (this.indexCache == null)
-			this.indexCache = new TreeMap<String, Index>();
-
-		Node[] nodes = client.getNodes();
-		for (Node node : nodes) {
-			if (node.isActive()) {
-				try {
-					String indexString = Info.request(getInfoPolicy(), node, "sindex");
-					if (!indexString.isEmpty()) {
-						String[] indexList = indexString.split(";");
-						for (String oneIndexString : indexList) {
-							Index index = new Index(oneIndexString);
-							this.indexCache.put(index.toKeyString(), index);
-						}
-					}
-					break;
-				} catch (AerospikeException e) {
-					log.error("Error geting Index informaton", e);
-				}
-			}
-		}
+	public void refreshIndexes() {
```

Comment:
As I understand synchronized was needed here because of `this.indexCache.put` while somebody could potentially retrieve items from the map. Since I replaced the logic with simple reference assignment with volatile modifier -- we do not need synchronized anymore.